Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h

```diff
@@ -658,11 +599,73 @@ namespace AWSCore
             AZ_Printf(logRequestsChannel, "Response Body:\n");
             PrintRequestOutput(responseContent);
         }
+
     public:
+        using OnSuccessFunction = AZStd::function<void(ServiceRequestJob* job)>;
+        using OnFailureFunction = AZStd::function<void(ServiceRequestJob* job)>;
+
         template<class Allocator = AZ::SystemAllocator>
         static ServiceRequestJob* Create(OnSuccessFunction onSuccess, OnFailureFunction onFailure = OnFailureFunction{}, IConfig* config = GetDefaultConfig())
         {
-            return azcreate(ServiceRequestJob, (onSuccess, onFailure, config), Allocator);
+            return azcreate(Function, (onSuccess, onFailure, config), Allocator);
+        }
+
+    };
+
+    /// A derived class that calls lambda functions on job completion.
+    template<class RequestTraits>
+    class ServiceRequestJob<RequestTraits>::Function
+        : public ServiceRequestJob
+    {
+    public:
+        // To use a different allocator, extend this class and use this macro.
+        AZ_CLASS_ALLOCATOR(Function, AZ::SystemAllocator, 0);
+
+        Function(OnSuccessFunction onSuccess, OnFailureFunction onFailure = OnFailureFunction{}, IConfig* config = GetDefaultConfig())
+            : ServiceRequestJob(false, config) // No auto delete - we'll take care of it
```

Comment:
is "we'll take care of it" referring to the fact we have [`DoCleanup()`](https://github.com/o3de/o3de/pull/7088/files#diff-d45732876461be69d2c6c718d6821d37cf7373f63a7dc969454491fb41e1f1d7R662) ?

Reply:
This comment is from our previous code and I'm reverting these two jobs to the old implementation. I'll check it and see whether I can make some cleanup for this comment. Thanks, Allie.

Reply:
The comment is probably unnecessary if you already commented in the overridden functions that it handles its own deletion.

Reply:
Just made some minor updates to this comment.